src/main/scala/no/ndla/draftapi/service/search/TagIndexService.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Part of NDLA draft-api.
+ * Copyright (C) 2020 NDLA
+ *
+ * See LICENSE
+ */
+
+package no.ndla.draftapi.service.search
+
+import com.sksamuel.elastic4s.http.ElasticDsl._
+import com.sksamuel.elastic4s.indexes.IndexRequest
+import com.sksamuel.elastic4s.mappings.MappingDefinition
+import com.typesafe.scalalogging.LazyLogging
+import no.ndla.draftapi.DraftApiProperties
+import no.ndla.draftapi.model.domain.Article
+import no.ndla.draftapi.model.search.{SearchableLanguageFormats, SearchableTag}
+import no.ndla.draftapi.repository.{DraftRepository, Repository}
+import org.json4s.native.Serialization.write
+
+trait TagIndexService {
+  this: SearchConverterService with IndexService with DraftRepository =>
+  val tagIndexService: TagIndexService
+
+  class TagIndexService extends LazyLogging with IndexService[Article, SearchableTag] {
+    implicit val formats = SearchableLanguageFormats.JSonFormats
+    override val documentType: String = DraftApiProperties.DraftTagSearchDocument
+    override val searchIndex: String = DraftApiProperties.DraftTagSearchIndex
+    override val repository: Repository[Article] = draftRepository
+
+    override def createIndexRequest(domainModel: Article, indexName: String): Seq[IndexRequest] = {
+      val tags = searchConverterService.asSearchableTags(domainModel)
+
+      tags.map(t => {
+        val source = write(t)
+        indexInto(indexName / documentType).doc(source).id(s"${t.language}.${t.tag}")
+      })
+    }
+
+    def getMapping: MappingDefinition = {
+      mapping(documentType).fields(
+        List(
+          textField("tag"),
+          keywordField("language")
```

Comment:
```
textField("tag"),
keywordField("language")
```
Hva er forskjellen mellom de to her, hva gjør at begge de to kan ikke være av samme type f.eks?

Reply:
textFields analyseres og tokenizes som gjør de søkbare, mens keywordFields indekseres som de er som gjør de raskere og sortere og filtrere på.
De kunne teknisk sett vært det samme (Og de kunne hatt [multifields](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/multi-fields.html) så de tilogmed ble begge), men søking på keywordFields er ikke helt heldig tror jeg, og samme med filtrering/sortering på textFields.

Elastic forklarer det ganske mye bedre enn meg i doc'en sin: https://www.elastic.co/guide/en/elasticsearch/reference/6.8/mapping-types.html